fix server migration after introducing multiple subnets

chuan137 opened the pull request:
Because of the new feature of multiple share network subnets, introduced
in antelope, Share server model has changed. Server migration driver
needs to be adjusted accordingly.

Should also be fixed upstream.

Change-Id: If94dbe7d4cf7b9a892b827cabe6edf019357c615

MatheusAndrade777 committed:
[NetApp] Fix share server migration

SVM Migration was failing using NetApp driver after changes on share server structure,
since now the share server can have more than one subnet.

This patch fix all the errors related to this change.

Closes-bug: #2018300
Change-Id: I9eae95ff3f66a3497b00ca582491afec58ae6dc3

kpawar-sap commented:
multiple subnet is fixed upstream a9344998347ec384417af19266fabb1f9490ce1b in maste. May you backport this commit to 2023.1-3 ?

chuan137 commented:
hmm, the commit did not fix the bug in `neutron_network_plugin.py`

> multiple subnet is fixed upstream [a934499](https://github.com/sapcc/manila/commit/a9344998347ec384417af19266fabb1f9490ce1b) in maste. May you backport this commit to 2023.1-3 ?

kpawar-sap commented:
ok, can you then modify this PR in two commits, one upstream as it is and then diff of yours (changes only related to neutron_network_plugin.py). This way we have minimum difference between upstream and sapcc code and easy to apply changes going forward.

chuan137 committed:
follow up "[NetApp] Fix share server migration"

This patch fixes extending network allocation (port binding) when
migrating across physical network.

Change-Id: I97c27fbf65320d6da3f3363da269f4d6264c802c

kpawar-sap commented:
one tempest test is failing 
manila.tests.network.neutron.test_neutron_plugin.NeutronBindNetworkPluginTest.test_extend_network_allocations
